### ISSUE TITLE
Error in Xamarin Studio 5.9

### ISSUE THREAD

**Issue by mteper** [mrward/monodevelop-nuget-extensions#2]
XS 5.9 shipped with an updated version of NuGet, causing this error to pop when attempting to manage NuGet packages for a solution:

```
System.IO.FileNotFoundException: Could not load file or assembly 'NuGet.Core, Version=2.8.1.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
File name: 'NuGet.Core, Version=2.8.1.0, Culture=neutral, PublicKeyToken=null'
  at MonoDevelop.PackageManagement.ManagePackagesHandler2.CreateViewModel () [0x00000] in <filename unknown>:0 
  at MonoDevelop.PackageManagement.ManagePackagesHandler2.Run () [0x00000] in <filename unknown>:0 
```

**Comment by mrward** [comment]
Xamarin Studio 5.9 was updated to NuGet 2.8.3 so the NuGet Extensions addin version 0.6 no longer works on the Mac.

I am currently unable to publish this to the main Xamarin Studio adding repository.

For now I have uploaded a 0.6.1 version of the addin to [GitHub](https://github.com/mrward/temp-addin-upload) which works with Xamarin Studio 5.9. Download the .mpack file, open the Add-in Manager and select Install from file to install the addin. You may need to restart after installing.

**Comment by mteper** [comment]
:+1: worked like a charm, thanks!!

**Comment by jchannon** [comment]
I've just updated to 5.9.3 (build 1) and 0.6.1 of the extension no longer works :cry:

**Comment by mrward** [comment]
Version 0.8 should work with Xamarin Studio 5.9.3. That version been published to the main Xamarin Studio Add-in Repository (Alpha Channel).

You can also download the version 0.8 mpack directly from [GitHub](https://github.com/mrward/monodevelop-addins/blob/gh-pages/5.0/MonoDevelop.PackageManagement.Extensions_0.8.mpack).

**Comment by jchannon** [comment]
Thanks! Did the trick

**Comment by mteper** [comment]
Any word on when this is coming natively to XS? Seems like a pretty fundamental feature...

**Comment by mrward** [comment]
Any particular part you would like to see included?

It is unlikely the NuGet PowerShell integration will be supported out of the box in the near future since the PowerShell implementation is incomplete.

Being able to manage packages at the solution level (i.e. being able to add, update, remove NuGet packages in multiple projects in one step) is something that I would like to see included out of the box in Xamarin Studio. But there is no word on when that will happen.

**Comment by mteper** [comment]
Yup, that's what I was talking about (ability to manage solution-wide). Most of my solutions have over a dozen projects. Managing dependency updates one at a time via UI is both a major PITA and error-prone. Your plugin is invaluable, but seems like it really belongs in the core product.

**Comment by jchannon** [comment]
Agreed

**Comment by alexrainman** [comment]
Not available anymore with XS 5.9.8

**Comment by mrward** [comment]
@alexrainman - By not available anymore with XS 5.9.8 I am assume you mean from the addin manager. I tried to publish version 0.10 of the addin which is compatible with the stable version of Xamarin Studio to the beta channel but for some reason addins.monodevelop.com is not showing it. It only seems to be showing 0.10.1 on the alpha channel which only works with XS 5.10.

You can download 0.10 of the addin from [GitHub](https://github.com/mrward/monodevelop-addins/blob/gh-pages/5.0/MonoDevelop.PackageManagement.Extensions_0.10.mpack). You can install the .mpack using the Install from file button in the Add-in Manager.

**Comment by alexrainman** [comment]
This is what i did:

1. Install mpack manually
2. Override dlls in XS Package Content folder

By the way, the extension is there in alpha channel.

Thanks

**Comment by Sposito** [comment]
I am trying to use it in MonoDevelop Unity Version (its a custom 5.9.6) but I am not having any luck. It is asking for MonoDevelop.PackMangament,5.0. Anyone here had any luck with Nuget and Unity?

**Comment by mrward** [comment]
@Sposito - Unity's version of MonoDevelop does not ship with the NuGet addin that the NuGet extensions depends on. You can see that if you open the Add-in Manager dialog.

As a quick test I downloaded the [beta version of Unity's MonoDevelop](http://forum.unity3d.com/threads/monodevelop-unity-5-9-preview-release.350016/) and it does not include the NuGet addin.

To get NuGet working I copied the MonoDevelop.PackageManagement directory from Xamarin Studio 5.9 into Unity's MonoDevelop Addins directory.

Getting the NuGet extensions to work may be problematic since you have to make sure the version of NuGet.Core is the same used by the main NuGet addin and the extensions addin, at least on the Mac, otherwise you will get the error reported in this bug report.

Also for some reason Unity's version of MonoDevelop does not set the correct MonoDevelop version. It is set to 5.9 but it should be set to 5.9.6. Unfortunately that means the only version of the NuGet extensions addin that you can install without any error is [version 0.7](https://github.com/mrward/monodevelop-addins/blob/gh-pages/5.0/MonoDevelop.PackageManagement.Extensions_0.7.mpack). So the correct version of the NuGet addin needs to be used so they are compatible. I have uploaded a [.zip](https://github.com/mrward/temp-addin-upload/tree/master/Unity) containing the binaries for the NuGet addin that should be compatible with NuGet extensions 0.7. This works on Windows with Unity's MonoDevelop 5.9.6.

My test steps were:

1. MonoDevelop not running.
2. Unzip the NuGet addin binaries into the Addins directory.
3. Run MonoDevelop and check that the NuGet dialog can be opened.
4. Install NuGet extensions 0.7 .mpack using the Add-in Manager.
5. Restart MonoDevelop - otherwise the PowerShell console is not available from the View menu.
6. Check that Manage Packages dialog can be opened and the PowerShell console can be opened.

It is most likely possible to get the latest version of the NuGet addin and the NuGet extensions to work but it would involve recompiling the NuGet extensions addin from source so Unity's MonoDevelop allows it to install or loaded.

**Comment by Sposito** [comment]
Great!! Now Manage Packages are working just fine!! Thanks a million!
About PowerShell I can't find it, where it is supposed to be?
Thanks again!

**Comment by AlejandroRuiz** [comment]
Im getting this with Xamarin Studio 5.10.1
<img width="423" alt="screen shot 2016-01-13 at 12 04 27 pm" src="https://cloud.githubusercontent.com/assets/1253565/12303171/827dde50-b9ee-11e5-9c88-1ca0dfaa5fc4.png">

**Comment by muer2000** [comment]
@mrward thanks!

**Comment by mfdeveloper** [comment]
@mrward I've got problems with latest Unity 2017.1.1f version, following the steps described in [issuecomment-170224450](https://github.com/mrward/monodevelop-nuget-extensions/issues/2#issuecomment-170224450). To solve this problem, I did the steps below: 

1. Extract `NuGet-Unity596` content to **Monodevelop** Unity `AddIns` folder.
2. Download `MonoDevelop.PackageManagement.dll` and `NuGet.Core.dll` from https://github.com/mrward/temp-addin-upload/tree/master/MonoDevelop.PackageManagement/2.8.7 and replace into: `[Monodevelop folder]/AddIns/MonoDevelop.PackageManagement` folder.
3. Download the file [MonoDevelop.PackageManagement.Extensions_0.10.1.mpack](https://github.com/mrward/temp-addin-upload/blob/master/MonoDevelop.PackageManagement.Extensions_0.10.1.mpack). 
4. Open monodevelop, click on `Tools => Add-in Manager` and click on button `Install from file`. Select the downloaded file above and install.
5. Click with right button on solution (or click on menu `Project`), and click on submenu `Manage Packages`.
5. Enjoy it! Search by nuget packages and install :)
